docs/components/bn-file-input.md

```diff
@@ -0,0 +1,326 @@
+          @click="addFile()"
+        >
+          Add File
+        </button>
+        <button
+          v-else
+          class="mb-2 rounded border border-gray-300 py-1 px-2 text-sm shadow"
+          @click="openFileDialog()"
+        >
+          Browse
+        </button>
+        <ul
+          v-if="value"
+          class="w-full"
+        >
+          <li
+            class="flex w-full items-center border border-t-0 p-1 text-sm first:border-t"
+          >
+            <img
+              :src="imagePreviewPath(value)"
+              class="mr-2 h-6 w-6 rounded-full"
+            >
+            <span class="truncate">{{ value.name }}</span> ({{ value.size / 1000 }} KB)
+            <button
+              class="ml-auto"
+              @click="removeFile(value)"
+            >
+              🗑
+            </button>
+          </li>
+        </ul>
+        </div>
+      </template>
+    </bn-file-input>
+  </div>
+</code-preview>
+
+### bottom
+
+Useful for hints or errors. Includes the following slot props:
+- `errorMessage`: `vee-validate` property, if the input is invalid
+- `valid`: `vee-validate` meta property
+- `touched`: `vee-validate` meta property
+
+```html
+<bn-file-input
+  v-model="file"
+  name="required"
+  :rules="isRequired"
+>
+  <template #bottom="{ errorMessage, valid, touched }">
+    <div
+      v-if="!valid && touched"
+      class="mt-1 flex items-center text-rose-700"
+    >
+      <svg
+        viewBox="0 0 24 24"
+        class="fill-none mr-1 h-4 w-4 stroke-current stroke-2"
+      >
+        <path
+          :d="warningIcon"
+        />
+      </svg>
+      <span class="mr-1">
+        {{ errorMessage }}
+      </span>
+      <svg
+        viewBox="0 0 24 24"
+        class="fill-none mr-1 h-4 w-4 stroke-current stroke-2"
+      >
+        <path
+          :d="warningIcon"
+        />
+      </svg>
+    </div>
+  </template>
+</bn-file-input>
+```
+
+<code-preview>
+  <div class="grid col-span-1 gap-4">
+    <bn-file-input
+      v-model="file"
+      name="required"
+      :rules="isRequired"
+    >
+      <template #bottom="{ errorMessage, valid, touched }">
+        <div
+          v-if="!valid && touched"
+          class="mt-1 flex items-center text-rose-700"
+        >
+          <svg
+            viewBox="0 0 24 24"
+            class="fill-none mr-1 h-4 w-4 stroke-current stroke-2"
+          >
+            <path
+              :d="warningIcon"
+            />
+          </svg>
+          <span class="mr-1">
+            {{ errorMessage }}
+          </span>
+          <svg
+            viewBox="0 0 24 24"
+            class="fill-none mr-1 h-4 w-4 stroke-current stroke-2"
+          >
+            <path
+              :d="warningIcon"
+            />
+          </svg>
+        </div>
+      </template>
+    </bn-file-input>
+  </div>
+</code-preview>
+
+
+## [TODO] Customization
+
+There are three ways to customize the appearance of the `BnFileInput` component:
```

Comment:
Acá faltaron las otras dos maneras. Igual esto quizás se podría dejar vacío o con TODO no más, porque es como lo mismo pa todos los componentes, quizás debería ir en otro lado?

Reply:
It would be nice to have this on another page if it's the same for all components. I will left this empty. I think that is the same for the color and theming section.